CodeGeneration/Sources/generate-swiftsyntax/SyntaxBaseNodesFile.swift

```diff
@@ -0,0 +1,235 @@
+          // to do a sanity check and verify the kind matches in debug builds and get
+          // maximum performance in release builds.
+          self.init(syntax._syntaxNode.data)
+        }
+        """)
+      
+      InitializerDecl("""
+        /// Create a `\(raw: node.name)` node from a specialized optional syntax node.
+        public init?<S: \(raw: node.name)Protocol>(_ syntax: S?) {
+          guard let syntax = syntax else { return nil }
+          self.init(syntax)
+        }
+        """)
+      
+      InitializerDecl("""
+        public init(fromProtocol syntax: \(raw: node.name)Protocol) {
+          // We know this cast is going to succeed. Go through init(_: SyntaxData)
+          // to do a sanity check and verify the kind matches in debug builds and get
+          // maximum performance in release builds.
+          self.init(syntax._syntaxNode.data)
+        }
+        """)
+      
+      InitializerDecl("""
+        /// Create a `\(raw: node.name)` node from a specialized optional syntax node.
+        public init?(fromProtocol syntax: \(raw: node.name)Protocol?) {
+          guard let syntax = syntax else { return nil }
+          self.init(fromProtocol: syntax)
+        }
+        """)
+      
+      InitializerDecl("public init?<S: SyntaxProtocol>(_ node: S)") {
+        SwitchStmt(expression: MemberAccessExpr("node.raw.kind")) {
+          SwitchCaseList {
+            SwitchCase(
+              label: .case(SwitchCaseLabel {
+                let childNodes = SYNTAX_NODES.filter { $0.baseKind == node.syntaxKind }
+                for childNode in childNodes {
+                  CaseItem(
+                    pattern: EnumCasePattern(
+                      period: .period,
+                      caseName: .identifier(childNode.swiftSyntaxKind)),
+                    trailingComma: childNode == childNodes.last ? nil : .comma)
+                }
+              })) {
+                Expr("self._syntaxNode = node._syntaxNode")
+              }
+            
+            SwitchCase("default:") {
+              ReturnStmt("return nil")
+            }
+          }
+        }
+      }
+      
+      InitializerDecl("""
+        /// Creates a `\(node.name)` node from the given `SyntaxData`. This assumes
+        /// that the `SyntaxData` is of the correct kind. If it is not, the behaviour
+        /// is undefined.
+        internal init(_ data: SyntaxData)
+        """) {
+        IfConfigDecl(
+          clauses: IfConfigClauseList {
+            IfConfigClause(
+              poundKeyword: .poundIfKeyword(),
+              condition: Expr("DEBUG"),
+              elements: IfConfigClauseSyntax.Elements.statements(CodeBlockItemList {
+                SwitchStmt(
+                  expression: Expr("data.raw.kind")) {
+                    SwitchCase(
+                      label: .case(SwitchCaseLabel {
+                        let childNodes = SYNTAX_NODES.filter { $0.baseKind == node.syntaxKind }
```

Comment:
Because we need to omit the last `,` so we don't have `.someCase, :` as it's not valid  swift syntax.

Reply:
It seems to me like `SwiftSyntaxBuilder` should handle this for us. This result builder is a `CaseItemListBuilder` and that contains `ensuringTrailingComma`. So you shouldn’t need to set a `trailingComma` on `CaseItem` at all. 

(same above in line 101 and below in line 206)